subprojects/androidTest/androidTest.gradle

```diff
@@ -18,10 +18,37 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
 
+    buildTypes {
+        debug {
+            enableUnitTestCoverage true
+            enableAndroidTestCoverage true
+        }
+    }
+    afterEvaluate {
+        // Enable coverage of mockito classes which are not part of this module.
+        // Without these added classFileCollection dependencies the coverage data/report is empty.
+        tasks.createDebugAndroidTestCoverageReport { com.android.build.gradle.internal.coverage.JacocoReportTask task ->
+            rootProject.allprojects { Project currentProject ->
+                plugins.withId("java") {
+                    SourceSetContainer sourceSets = currentProject.sourceSets
+                    // Mimic org.gradle.testing.jacoco.tasks.JacocoReportBase.sourceSets().
+                    // Not possible to set task.javaSources because it's initialized with setDisallowChanges.
+                    //task.javaSources.add({ [ sourceSets.main.allJava.srcDirs ] })
+                    task.classFileCollection.from(
+                        sourceSets.main.output.asFileTree.matching {
+                            exclude("**/internal/util/concurrent/**")
+                        }
+                    )
```

Comment:
Do we need to exclude this?
```suggestion
                    task.classFileCollection.from(sourceSets.main.output)
```

From an earlier run it looked like it's not used by these tests:
![image](https://github.com/mockito/mockito/assets/2906988/c7d29841-d314-440a-ab10-7f168d5bfa99)

Reply:
Why are they excluded in the first place in the normal report?

Reply:
The classes in that folder are from a different library that we bundle. See the license (https://github.com/mockito/mockito/blob/222c14c97bc2ec59ec5e486ddaa6438edded0e5c/src/main/java/org/mockito/internal/util/concurrent/LICENSE) and corresponding readme. Since we repackage this library, we don't include it in our code coverage.

Reply:
From a previous CI run (screenshot above) it looks like the androidTest is not exercising any lines from it. I wonder if the maintenance of those excludes is worth the minor percentage difference gain for the test coverage. Anyway, it works now.